persist/fs/msgpack/schema.go

```diff
@@ -82,4 +84,5 @@ func init() {
 	setNumFieldsForType(logInfoType, numLogInfoFields)
 	setNumFieldsForType(logEntryType, numLogEntryFields)
 	setNumFieldsForType(logMetadataType, numLogMetadataFields)
+	setNumFieldsForType(indexInfoTypeLegacyV1, numIndexInfoFieldsLegacyV1)
```

Comment:
Hm if we went ahead and made this change, why do we still need the `allowSkip`?  Can't we just detect the type and decode accordingly?

Reply:
We can't because its not forward-compatible. I.E the old binary will try and read the new files and then complain about the wrong object type